### DISCUSSION TITLE
GSNR Measurement in Multi-Band system

### DISCUSSION THREAD

**Discussion by MirkoZETA** [Telecominfraproject/oopt-gnpy#514] [Q&A] [answered]
Hello Team and Community,

I'm relatively new to GNPY, and I'm interested in using it to calculate the GSNR of a given lightpath as a performance metric in a Multi-Band system, specifically for C+L+S and C+L+E bands. I understand that setting the spectrum is fairly straightforward, but I’m curious if GNPY currently supports systems beyond C+L.

As mentioned in issue #508 by @EstherLerouzic "the fiber modelling properly supports multiband case, with proper SRS interband power transfer, the current release does not support the modelling of C+L amps yet, so that the same Edfa is used for both bands". Is this still the case?

My goal is to calculate this metric for a given topology and determine the feasibility of certain lightpaths. Any guidance or information on this topic would be greatly appreciated.

Thank you for your great work!

**Comment by AndreaDAmico** [comment, marked as answer]
Dear @MirkoZETA,

The current version of GNPy does not support the modeling of C+L amps yet, we are currently reviewing the new patches integrating this features and we expect to merge them in the next couple of months.

In the mean time, as a workaround, you can try to set a constant gain in the EDFA and shape the actual gain profile on multiple bands using the EDFA parameter `gain_ripple` (same concept works for the noise figure). As already mentioned, the fiber modeling already support multi-band transmission. It is crucial to activate in this case the Raman solver, setting `True` the flag in `sim_params.raman_solver`.

Also,  if available, a precise set of fiber parameters can be provided to GNPy as profiles along frequency. This will increase the accuracy of the simulation, especially in wideband transmission scenarios.

I hope this may help.

**Comment by MirkoZETA** [comment]
Thank you for the detailed response; I’ll review it carefully. By the way, I noticed a multiband equipment file and some example setups, so I assume these features are still under development. I look forward to future updates and improvements in this area, as multi-band support would be invaluable for my use case.

Thanks again for your response!
